File: JavaScript/examples/example_triggered_acquisition.js
// Example: triggered acquisition (external trigger on TRIG input)
// Requirements: npm install serialport

const PORT = 'COM3'; // change to your COM port
const FRAMES = 1000;
const TRIGGER_RISING = true; // false for falling edge

const { CoreDAQ } = require('../coredaq_js_api');

(async () => {
  const dev = await CoreDAQ.open(PORT);
  try {
    await dev.arm_acquisition(FRAMES, true, TRIGGER_RISING);
    console.log('Armed. Waiting for trigger...');
    await dev.wait_for_completion(0.25, 10);

    const mv = await dev.transfer_frames_mV(FRAMES);
    console.log('CH1 first 10 samples (mV):', mv[0].slice(0, 10));
  } finally {
    await dev.close();
  }
})();

File: Python/examples/example_triggered_acquisition.py
# Example: triggered acquisition (external trigger on TRIG input)
# Requirements: pip install pyserial

PORT = "COM3"  # change to your COM port
FRAMES = 1000
TRIGGER_RISING = True  # False for falling edge

import os, sys
sys.path.append(os.path.dirname(os.path.dirname(__file__)))
from coredaq_python_api import CoreDAQ

dev = CoreDAQ(PORT)
try:
    dev.arm_acquisition(FRAMES, use_trigger=True, trigger_rising=TRIGGER_RISING)
    print("Armed. Waiting for trigger...")
    dev.wait_for_completion(timeout_s=10)

    mv = dev.transfer_frames_mV(FRAMES)
    print("CH1 first 10 samples (mV):", mv[0][:10])
finally:
    dev.close()